---
title: "UT_B117_main"
output: html_document
---
This script loads in the raw data files and calls the functions that run the logistic growth model of B117 vs wildtype and the SEIR model the projects infections
```{r setup, include=FALSE}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(dplyr)
library(EpiEstim)
library(tidyr)
library(EnvStats)
library(reshape2)
library(cowplot)
library(knitr)
library(flexmix)
library(zoo)
library(DescTools)
library(dplyr)
library(tidyverse)
library(lubridate)
library(lme4)
library(rstanarm)
library(ggpubr)
options(mc.cores = parallel::detectCores())
library(cowplot)
source("UT_B117_functions.R")
theme_set(theme_minimal_grid() + theme(legend.position="top"))


```
Load in SGTF data and PCT case data (not publicly available)
```{r}
DATA_PATH_CTS<-'../data/pctData_posOnly_210505.csv'
DATA_PATH_CASES<-'../data/UT_data_streams.csv'
```
Load in introduction estimates (publicly available)
```{r}
DATA_PATH_INTROS<-'../data/introductions_Jan15.csv'
```
Dates and parameters for analysis
```{r}
spring_arrival_date<-as.Date("2021-01-16", format = "%Y-%m-%d")
spring_last_date<-as.Date("2021-04-09", format = "%Y-%m-%d")
spring_end_semester_date<-as.Date("2021-05-23", format = "%Y-%m-%d")
initial_estimate_date<-as.Date("2021-02-12", format = "%Y-%m-%d")
fall_arrival_date<-as.Date("2020-08-20", format = "%Y-%m-%d")
fall_last_date<-as.Date("2020-12-19", format = "%Y-%m-%d")

# Some global parameters
QC_thres<-28
delta<-1/7 #(1/duration of infectiousness)
gamma<- 1/3 # (1/duration of latent period)
dt<-0.1
mean_GI<-5.8 # average generation interval (for EpiEstim)
std_GI<-2.3 # std dev generation interval (for EpiEstim)
N_POPULATION <-30000 # estimated "active" population at UT
prop_prev_inf_Aug = 0.05 
nsamps<-500 # number of samples from the distributions for uncertainty
# set prior beta values before we had any sequencing
a1<-40 
b1<-10
# sequencing results 
a2<-22.5 # 0.5 + NB117 sequenced
b2<-1.5 # 0.5 + NSGTF - N B117 sequenced
beta_pars<-data.frame(a1, b1)
nsim <- 100 #Number of Monte Carlo draws
```
Process the Ct data to get SGTF and non-SGTF positives from the Ct values of the 3 target genes (uses data not publicly available, see next chunk to load in summary outputs)
```{r}
Ct_data<-read.csv(DATA_PATH_CTS) %>%
  mutate(test_date = as.Date(test_date))
Ct_data<-Ct_data %>%select(test_date, N.gene.Ct, S.gene.Ct, ORF1ab.Ct)%>%
  filter(test_date>= spring_arrival_date)%>%
  filter(test_date<=spring_last_date)
Ct_data<-Ct_data[order(Ct_data$test_date),]
Ct_data<-Ct_data%>%rowwise()%>%mutate(mean_Ct = mean(c(N.gene.Ct, S.gene.Ct, ORF1ab.Ct), na.rm = TRUE))
Ct_data['S_dropout']<-is.na(Ct_data$S.gene.Ct)
Ct_data$poss_B117<-Ct_data$S_dropout==TRUE & Ct_data$N.gene.Ct<QC_thres
Ct_data['strain']<-rep(NA, nrow(Ct_data))
for (i in 1:nrow(Ct_data)){
  if (Ct_data$S_dropout[i] == TRUE){
    Ct_data$strain[i]<-'SGTF'
  }
  else 
    Ct_data$strain[i]<-'non_SGTF'
}

#Make a day to day summary of SGTF presence
Ct_data_daily_long<-Ct_data%>%filter(mean_Ct<QC_thres)%>%
  group_by(test_date, strain)%>%summarise(n = n())
Ct_data_daily_summary<-Ct_data_daily_long%>%
  pivot_wider(names_from = strain, values_from = n, values_fill = 0)
ut<-Ct_data_daily_summary%>%group_by(test_date)%>%
  mutate(n_Sdrop = SGTF,
         n_Spos = non_SGTF,
         n_pos = SGTF+non_SGTF,
         prop = SGTF/n_pos,
         prop_se = sqrt(prop*(1-prop)/n_pos))
# Export the daily data for fitting. Will use this to feed into Woody's SGTF.R code
write.csv(Ct_data_daily_summary, '../data/Ct_data.csv')
# Make a dataset used for the initial fitting from data through Feb.
Ct_data_init<-Ct_data_daily_summary%>%filter(test_date<=initial_estimate_date)
write.csv(Ct_data_init, '../data/Ct_data_init.csv', row.names = FALSE)

path ="../out/dailySGTF.png"
ut %>%
  filter(n_pos >= 1) %>% 
  ggplot() +
  geom_linerange(aes(test_date, ymin = prop - prop_se, ymax = prop + prop_se)) + 
  geom_point(aes(test_date, prop)) +
  scale_y_continuous(labels=scales::percent) + 
  labs(x = "Test Collection Date", y = "Frequency of SGTF among cases") 
ggsave(filename = path, device = "png", dpi = 700) 

path = "../out/stackedbarSGTF.png"
stacked<-ggplot(Ct_data_daily_long, aes(x = test_date, y = n, fill = strain))  + geom_bar(position = "stack", stat = "identity") + xlab('Test Collection Date') + ylab('Number of low Ct positives') + theme(legend.position="top")+ scale_fill_discrete(name = "S gene presence", labels = c("SGTF", "non-SGTF"))
stacked
ggsave(filename = path, device = "png", dpi = 700)



#Use SGTF data set to call function to fit the logistic model with uncertainty from SGTF presence. 


#Run the function which fits the logistic growth model, using data through Feb 12 and April 9th
posterior_list<-fit_SGTF_data(ut, beta_pars, spring_arrival_date, spring_end_semester_date, nsim)

plotdf<-posterior_list[[6]]

#output: c and k in logistic growth model (intercept and growth rate), current B117 prevalence
# Early estimate: Jan 16- Feb. 12 (probably won't need this)
c1_distrib<-posterior_list[[1]]
k1_distrib<-posterior_list[[2]]


# Use later estimate (Jan 16- April 9) for p(t) and transmissibiltiy advantage
c_distrib<-posterior_list[[3]]
k_p_distrib<-posterior_list[[4]]
p_curr_distrib<-posterior_list[[5]]



# Write the estimated distributions into csv files (to be loaded in later)
write.csv(plotdf, '../data/fig1adf.csv', row.names = FALSE)
write.csv(c_distrib, '../data/c_distrib.csv', row.names = FALSE)
write.csv(k_p_distrib,'../data/k_p_distrib.csv', row.names = FALSE)
write.csv(p_curr_distrib, '../data/p_curr_distrib.csv', row.names = FALSE)

# Transmissibility advantage based on estimates 
M1_distrib = exp(mean_GI*k1_distrib)
M2_distrib = exp(mean_GI*k_p_distrib) # This should probably be a function of b0 and b1? 
M_CI<-quantile(M2_distrib, probs = c(0.025, 0.5, 0.975))


## Weekly summaries of prevalence
ut_weekly <- ut %>%
  mutate(eweek = epiweek(test_date),
         weekday = wday(test_date, label=TRUE)) %>%
  group_by(eweek) %>%
  summarize(prop = sum(n_Sdrop) / sum(n_pos),
            n = sum(n_pos),
            test_date = test_date[weekday == "Wed"],
            prop_se = sqrt(prop * (1 - prop) / n)) %>%
  mutate(dataset = ifelse(test_date <= initial_estimate_date,
                          "Feb 12",
                          "Apr 9")) %>%
  glimpse()

write.csv(ut_weekly, '../data/ut_weekly.csv', row.names = FALSE)

```

Load in the summary dataframe for making Fig 1A
```{r}
plotdf<-read.csv('../data/fig1adf.csv')
ut_weekly<-read.csv('../data/ut_weekly.csv')
plotdf<-plotdf%>%mutate(test_date = ymd(test_date))
ut_weekly<-ut_weekly%>%mutate(test_date = ymd(test_date))

## Quantiles of beta distribution (for expanding confidence intervals)
myqs <- qbeta(c(0.025, 0.975, 0.5), a2, b2)

path = "../out/comp_weekly.png"
comp_plot2 <- plotdf %>%
  ggplot() +
  ## Prevalence curve estimate and interval for Feb 12 report
  geom_ribbon(aes(test_date, ymin=pred_lo, ymax=pred_hi, fill = "Feb 12"), alpha=0.2,size=0.5) +
  geom_line(aes(test_date, pred, col = "Feb 12"), size=0.75) +
  ## Prevalence curve estimate and interval for updated Apr 9 report
  geom_ribbon(aes(test_date, ymin=pred2_lo, ymax=pred2_hi, fill = "Apr 9"), alpha=0.2,size=0.5) +
  geom_line(aes(test_date, pred2, col = "Apr 9"), size=0.75) +
  ## Point estimates and 95% confidence intervals for test sample prevalence
  ## (i.e. from data)
  geom_linerange(data = ut_weekly,
                 aes(test_date,
                     ymin = pmax(0, myqs[1] * (prop - 1.96 * prop_se)),
                     ymax = pmin(1, myqs[2] * (prop + 1.96 * prop_se)))) +
  geom_point(data = ut_weekly %>% filter(n>=1),
             aes(test_date, prop * myqs[3],
                 size=n,
                 ## shape = dataset,
                 color = dataset,
                 )) +
  ## Adjust the scales
  scale_y_continuous(labels=scales::percent) +
  scale_color_manual("Estimated growth", values=c("dodgerblue3", "darkgreen"),
                     breaks = c("Feb 12", "Apr 9")) +
  scale_fill_manual("Estimated growth", values=c("dodgerblue3", "darkgreen"),
                    breaks = c("Feb 12", "Apr 9")) +
  scale_shape_manual("Estimated prevalence", values=c(19, 1),
                     breaks = c("Feb 12", "Apr 9")) +
  scale_size("Sample size") +
  scale_x_date(date_labels="%b %d", limits = c(spring_arrival_date, spring_end_semester_date)) +
  ## Axis labels
  labs(
    x = "",
    y = "Frequency of B.1.1.7 among cases"## ,
       ## caption = "Source: UT-Austin HTTC"
  )  +
  ## Adjust details
  theme(legend.position="top",
        ## legend.title = element_text(face="bold"),
        legend.direction = "horizontal",
        ## axis.title = element_text(face="bold")
        )+
  guides(shape = guide_legend(title.position = "top"),
         size = guide_legend(title.position = "top"),
         color = guide_legend(title.position = "top"),
         fill = guide_legend(title.position = "top")) +
  NULL

comp_plot2

ggsave(filename = path, device = "png", dpi = 700)

```



Load in the logistic growth model outputs (posterior distributions of the y-intercept and growth rate, and the current estimate of the B117 prevalence)** These will be used as inputs into the SEIR model**
```{r}

k_p_distrib<-read.csv('../data/k_p_distrib.csv')
k_p_distrib<-k_p_distrib$x
c_distrib<-read.csv('../data/c_distrib.csv')
c_distrib<-c_distrib$x
p_curr_distrib<-read.csv('../data/p_curr_distrib.csv')
p_curr_distrib<-p_curr_distrib$x

# From the literature (Davies 2021 US estimate of 59% more transmissibile (56-63))
k_tr_adv_distrib<-rtri(n = nsamps, min = log(1.56)/5.5, max = log(1.63)/5.5, mode = log(1.59)/5.5)
# find literature CI on growth rate
k_lit_CI<-quantile(k_tr_adv_distrib, probs = c(0.025, 0.5, 0.975))

# Median estimates of yintercept and growth rate from local estimate. 
c = quantile(c_distrib, probs = 0.5)
p_curr<-quantile(p_curr_distrib, probs = 0.5)
k_p = quantile(k_p_distrib, probs = 0.5)
k_p_CI = quantile(k_p_distrib, probs = c(0.025, 0.975))
p_curr_CI = quantile(p_curr_distrib, probs = c(0.025, 0.975))

```

Load in the UT case data, clean and estimate Rt (skip to next chunk for Rt summaries)
```{r}
case_data<-read.csv(DATA_PATH_CASES)%>%mutate(date = as.Date(CollectionDate, format = "%Y-%m-%d"))%>%
                                                select(date, student_pos, student_PCT_pos,
                                                       total_student_PCT_tests)%>%
                                                filter(date>=fall_arrival_date, date<=spring_last_date+7)
# Add some columns to the case data
case_data<-case_data%>%mutate(positivity = student_PCT_pos/total_student_PCT_tests)
sevendayPCTpos = c(rep(NA,3), rollapply(case_data$positivity, 7, function(x) mean(x, na.rm = TRUE)), rep(NA,3))
sevendayPCTcases = c(rep(NA,3), rollapply(case_data$student_PCT_pos, 7, function(x) mean(x, na.rm = TRUE)), rep(NA,3))
sevendaycases = c(rep(NA,3), rollapply(case_data$student_pos, 7, function(x) mean(x, na.rm = TRUE)), rep(NA,3))
case_data$sevendayPCTpos = sevendayPCTpos
case_data$sevendayPCTcases = sevendayPCTcases
case_data$sevendaycases = sevendaycases

case_data%>%ggplot() + geom_line(aes(x = date, y = sevendayPCTpos)) + geom_bar(aes(x = date, y = positivity), stat = "identity") + ylab('PCT positivity') + xlab('Test Collection Date')

case_data%>%ggplot() + geom_line(aes(x = date, y = sevendayPCTcases)) + geom_bar(aes(x = date, y = student_PCT_pos), stat = "identity") + ylab('Student PCT cases') + xlab('Test Collection Date')

# Estimate R(t) in the fall and in the spring.
#daily_p_local<-get_daily_p_local(case_data, fall_arrival_date)
case_data_fall<-case_data%>%filter(date<=fall_last_date)
case_data_spring<-case_data%>%filter(date>=spring_arrival_date)
cases_fall<-case_data_fall$student_pos # for fall, we use total positives

avg_tests_per_day<-mean(case_data_spring$total_student_PCT_tests, na.rm = TRUE)
case_data_spring$positivity[is.na(case_data_spring$positivity)]=case_data_spring$sevendayPCTpos[is.na(case_data_spring$positivity)]
# Still need to do something to account for 10-day pause during the Texas winter freeze where no tests were collected, use the 3 days before and after it to impute. 
case_data_around_freeze<-case_data%>%filter(date<=as.Date("2021-02-24"), date>=as.Date("2021-02-10"),
                                            total_student_PCT_tests>0)
avg_positivity_freeze<-sum(case_data_around_freeze$student_PCT_pos)/sum(case_data_around_freeze$total_student_PCT_tests)
avg_n_cases_freeze<-sum(case_data_around_freeze$student_pos)/nrow(case_data_around_freeze)
case_data_spring$positivity[is.na(case_data_spring$positivity)]=avg_positivity_freeze
cases_spring<-round(avg_tests_per_day*case_data_spring$positivity,0) # for spring, we used PCT positivity
case_data_spring$imputed_cases<-cases_spring


Rt_summary_fall<-Rt_fxn_cases(cases_fall, case_data_fall, fall_arrival_date, fall_last_date, mean_GI,std_GI)
Rt_summary_spring<-Rt_fxn_cases(cases_spring, case_data_spring, spring_arrival_date, spring_last_date+7, mean_GI,std_GI)

# Export the Rt data for use in the SEIR model and the imputed cases
write.csv(Rt_summary_fall, '../data/Rt_students_fall.csv', row.names = FALSE)
write.csv(Rt_summary_spring, '../data/Rt_students_spring.csv', row.names = FALSE)
write.csv(case_data_spring, '../data/case_data_spring.csv', row.names = FALSE)

path = "../out/student_cases.png"
cases<-ggplot(case_data_spring) + geom_line(aes(x = date, y = sevendayPCTcases)) + geom_bar(aes(x = date, y = student_PCT_pos), stat = "identity") + xlab('Date') + ylab('Reported PCT cases') 
cases
ggsave(filename = path, device = "png", dpi = 700)
tsim<-seq(from = 0, to = length(case_data_spring$date)-1, by=1)
p_t <-(1/(1+exp(-(c+k_p*tsim))))
case_data_spring$n_est_B117<-case_data_spring$student_PCT_pos*p_t
case_data_spring$n_est_WT<-case_data_spring$student_PCT_pos*(1-p_t)
case_data_long<-case_data_spring%>%select(date, n_est_B117, n_est_WT)%>%
  pivot_longer(cols = starts_with("n_est_"), names_to = "strain", names_prefix = "n_est_", values_to = "n", values_drop_na = TRUE)

# Imputed cases that are B117 vs wildtype
path = "../out/student_cases_by_B117.png"
cases<-ggplot(case_data_long, aes(x = date, y = n, fill = strain)) + geom_bar(position = "stack", stat = "identity") + xlab('Date') + ylab('Reported PCT cases') + scale_fill_manual( name = "variant", labels = c("B.1.1.7", "wildtype") , values = c("aquamarine4", "coral1")) + theme(legend.position = "right")
cases
ggsave(filename = path, device = "png", dpi = 700)



```
Load in Rt estimates from fall and spring and imputed cases to generate figures 
```{r}
Rt_summary_fall<-read.csv('../data/Rt_students_fall.csv')
Rt_summary_spring<-read.csv('../data/Rt_students_spring.csv')
Rt_summary_fall$dates<-as.Date(Rt_summary_fall$dates)
Rt_summary_spring$dates<-as.Date(Rt_summary_spring$dates)
case_data_spring<-read.csv('../data/case_data_spring.csv')
case_data_spring$date<-as.Date(case_data_spring$date)

Rt_summary_fall%>%ggplot() + geom_line(aes(x =dates, y = Rt_medians))+ geom_ribbon(aes(x = dates, ymin = Rt_lowerbounds, ymax = Rt_upperbounds), alpha = 0.1)

Rt_summary_spring%>%ggplot() + geom_line(aes(x =dates, y = Rt_medians))+ geom_ribbon(aes(x = dates, ymin = Rt_lowerbounds, ymax = Rt_upperbounds), alpha = 0.1)


# Just look at the median prevalence of B.1.1.7 over time
tspring<-seq(from = 0, to = nrow(Rt_summary_spring)-1, by =1)
p_t<-(1/(1+exp(-(c+k_p*tspring))))
plot(tspring, p_t)



#Run the SEIR model using the fall Rt estimate and fall intros to estimate the number previously infected

fall_intros<-read.csv(DATA_PATH_INTROS)%>%filter(Location == "total", Time == "Fall")%>%select(ML_infections_lo, ML_infections_med, ML_infections_hi)
I0bounds<-c(fall_intros$ML_infections_lo, fall_intros$ML_infections_hi)

par_table<-data.frame(dt, delta, gamma, N_POPULATION, prop_prev_inf_Aug, nsamps)

cumIJan<-run_fall_SEIR(par_table, I0bounds, Rt_summary_fall)
cumICI<-(cumIJan/N_POPULATION)%>%quantile(probs = c(0.025, 0.5, 0.975))
print(cumICI)
```
Use the previous infection estimates from the fall, spring Rt, the current estimates of variant prevalence, and variant increase in transmissibility (from both model estimates) to project infections with and without the variant.
```{r}

spring_par_table<-data.frame(dt, delta, gamma, N_POPULATION, nsamps)
iRts<-c(18, 14) # indices in spring Rt that correspond to the faster and slower spread scenarios (chosen retrospectively)

# Output from logistic model is used for B117 prevalence timeline and transmissibility advantage
out_list<-run_spring_SEIR(spring_par_table, case_data_spring, Rt_summary_spring, iRts, spring_last_date, cumIJan, k_p_distrib, k_p_distrib, c_distrib, p_curr_distrib)

df_t_sim<-out_list[[1]] # contains the time course of runs for 500 simulations for both scenarios
df_t<-out_list[[2]] # contains the median and bounds of the time course runs for both scenarios
df_summary<-out_list[[3]] # contains all the summary statistics for static variables

# Run again, but this time use literature for transmissibility advantage
out_list<-run_spring_SEIR(spring_par_table, case_data_spring, Rt_summary_spring, iRts, spring_last_date, cumIJan, k_tr_adv_distrib, k_p_distrib, c_distrib, p_curr_distrib)



df_t_sim2<-out_list[[1]] # contains the time course of runs for 500 simulations for both scenarios
df_t2<-out_list[[2]] # contains the median and bounds of the time course runs for both scenarios
df_summary2<-out_list[[3]] # contains all the summary statistics for static variables
df_t_sim['I_variant_lit']<-df_t_sim2$I_variant
df_t['Imedian_lit']<-df_t2$Imedianv

```

Make some plots of the infection projection outputs
```{r}
#Factor increase in transmission rate over time
path = "../out/factor_inc_t.png"
plt_ft<-ggplot() + geom_line(data = df_t, aes (x = springdates, y = f_t_median)) + geom_ribbon(data = df_t, aes(x = springdates, ymin = f_t_lb, ymax = f_t_ub), alpha = 0.1) +geom_line(data = df_t2, aes (x = springdates, y = f_t_median), color = 'red') + geom_ribbon(data = df_t2, aes(x = springdates, ymin = f_t_lb, ymax = f_t_ub), alpha = 0.1, fill = "red")+
  geom_vline(xintercept = as.Date(spring_last_date))+ xlab('Time') + ylab('Multiplicative factor on transmission rate')
plt_ft

N<-N_POPULATION
case_data_spring['calib']<-rep('fit', nrow(case_data_spring))
for(i in 1:nrow(case_data_spring)){
  if (case_data_spring$date[i]>spring_last_date){
    case_data_spring$calib[i]<-'observed'
  }
}

ggsave(filename = path, device = "png", dpi = 700)

# Infection estimates alongside observed test positivity
case_data_spring<-case_data_spring%>%filter(date<=spring_last_date)
path = "../out/both_scenarios.png"
plot1<-ggplot() + geom_line(data = df_t_sim, aes(x = dates, y = 1000*I_variant/N, group = samples), alpha = 0.05, color = "tan1") + geom_line(data = df_t_sim, aes(x = dates, y = 1000*I_variant_lit/N, group = samples), alpha = 0.03, color = "darkolivegreen4")+
  geom_line(data = df_t_sim, aes(x = dates, y = 1000*I_novariant/N, group = samples), alpha = 0.03, color = "mediumpurple") +
  geom_line(data = df_t, aes(x = springdates, y = 1000*Imediannv/N), size = 1, color = "mediumpurple")+ geom_line(data = df_t, aes(x = springdates, y = 1000*Imedianv/N), size = 1, color = "tan1") +geom_line(data = df_t, aes(x = springdates, y = 1000*Imedian_lit/N), size = 1, color = "darkolivegreen4")+
  geom_point(data = case_data_spring, aes(x = date, y = 1000*sevendayPCTpos), size = 0.5) +facet_wrap(~scenario) + 
  xlab ("Date") + ylab('COVID-19 infections per 1000')+theme(strip.background = element_blank(), aspect.ratio = 0.8) + coord_cartesian(ylim = c(0, 50))
plot1
ggsave(filename = path, device = "png", dpi = 700)

df_fast<-df_t_sim%>%filter(scenario=="Faster spread")
df_t_fast<-df_t%>%filter(scenario == "Faster spread")
path ="../out/fast_spaghettis.png"
fast_only<-ggplot() + geom_line(data = df_fast, aes(x = dates, y = 1000*I_variant/N, group = samples), alpha = 0.05, color = "tan1") +
  geom_line(data = df_fast, aes(x = dates, y = 1000*I_novariant/N, group = samples), alpha = 0.03, color = "mediumpurple") +
   geom_line(data = df_fast, aes(x = dates, y = 1000*I_variant_lit/N, group = samples), alpha = 0.03, color = "darkolivegreen4")+
  geom_line(data = df_t_fast, aes(x = springdates, y = 1000*Imediannv/N), size = 1, color = "mediumpurple")+ geom_line(data = df_t_fast, aes(x = springdates, y = 1000*Imedianv/N), size = 1, color = "tan1") +geom_line(data = df_t_fast, aes(x = springdates, y = 1000*Imedian_lit/N), size = 1, color = "darkolivegreen4")+
  geom_point(data = case_data_spring, aes(x = date, y = 1000*sevendayPCTpos), size = 1) + scale_x_date(date_labels="%b %d")+
  xlab ("Date") + ylab("") +theme(strip.background = element_blank())+ coord_cartesian(ylim = c(0, 50)) + ylab('COVID-19 infections per 1000') + xlab ("Date")
fast_only
ggsave(filename = path, device = "png", dpi = 1000, width = 7, height = 5, units="in")
save.image(file="foobar.Rdata")
```
Uses B117 prevalence figure from above to combine 
```{r}
path ="../out/figure1.png"
figure1<-ggarrange(fast_only, comp_plot2, ncol = 1, nrow = 2, common.legend = TRUE, legend = "right")
figure1
ggsave(filename = path, device = "png", dpi = 1000, width = 7, height = 10, units="in")

```
```{r}
# all 4 scenarios median model lines
df_imputed<-df_t%>%filter(springdates<=as.Date(spring_last_date))
path = "../out/all4medians.png"
plot4<-ggplot() + geom_line(data = df_t, aes(x = springdates, y = 1000*Imediannv/N, group = scenario, linetype = scenario),color = "mediumpurple", size = 1) + 
  geom_line(data = df_t, aes(x = springdates, y = 1000*Imedianv/N, group = scenario, linetype = scenario), color = "tan1", size = 1) +geom_line(data = df_t, aes(x = springdates, y = 1000*Imedian_lit/N, group = scenario, linetype = scenario), color = "darkolivegreen4", size = 1)+ geom_line(data = df_imputed, aes(x = springdates, y = 1000*Imedianv/N), color = "black", size = 2)+
  xlab ("Date") + ylab('COVID-19 infections per 1000')+theme(aspect.ratio = 0.8) + coord_cartesian(ylim = c(0, 50))+ ggtitle("Median model lines by scenario")
plot4
ggsave(filename = path, device = "png", dpi = 700)



# imputed/projected infections B117 and WT
path = "../out/both_inf_t.png"

plot_both<-ggplot() + geom_line(data = df_t, aes(x=springdates, y = 1000*B117medianv/N), color = "red", linetype = "dotted", size = 1) + geom_line(data = df_imputed, aes(x=springdates, y = 1000*B117medianv/N), color = "red", size = 1)+
  geom_ribbon(data = df_t, aes(x = springdates, ymin = 1000*B117lbv/N, ymax = 1000*B117ubv/N), alpha = 0.1, fill = "red", linetype = "dotted", size = 1) +
  geom_line(data = df_t, aes(x=springdates, y = 1000*WTmedianv/N), color = "blue", linetype = "dotted", size = 1)+geom_line(data = df_imputed, aes(x=springdates, y = 1000*WTmedianv/N), color = "blue", size = 1)+
  geom_ribbon(data = df_t, aes(x = springdates, ymin = 1000*WTlbv/N, ymax = 1000*WTubv/N), alpha = 0.1, fill = "blue")+
  geom_line(data = df_t, aes(x=springdates, y = 1000*Imedianv/N), color = "black", size = 1, linetype = "dotted")+
  geom_line(data = df_imputed, aes(x=springdates, y = 1000*Imedianv/N), color = "black", size = 1)+
  geom_ribbon(data = df_t, aes(x = springdates, ymin = 1000*Ilbv/N, ymax = 1000*Iubv/N), alpha = 0.1, fill = "black")+coord_fixed(50)+scale_y_log10()+
  facet_wrap(~scenario) + xlab ("Date") + ylab('Estimated') + theme(axis.text.x = element_text(angle=0))
plot_both
ggsave(filename = path, device = "png", dpi = 700)
path = "../out/both_inf_t_lit.png"
df_imputed2<-df_t2%>%filter(springdates<=spring_last_date)
plot_both_lit<- ggplot() + geom_line(data = df_t2, aes(x=springdates, y = 1000*B117medianv/N), color = "red", linetype = "dotted", size = 1) + geom_line(data = df_imputed2, aes(x=springdates, y = 1000*B117medianv/N), color = "red", size = 1) +
  geom_ribbon(data = df_t2, aes(x = springdates, ymin = 1000*B117lbv/N, ymax = 1000*B117ubv/N), alpha = 0.1, fill = "red") +
  geom_line(data = df_t2, aes(x=springdates, y = 1000*WTmedianv/N), color = "blue", size = 1, linetype = "dotted")+geom_line(data = df_imputed2, aes(x=springdates, y = 1000*WTmedianv/N), color = "blue", size = 1)+
  geom_ribbon(data = df_t2, aes(x = springdates, ymin = 1000*WTlbv/N, ymax = 1000*WTubv/N), alpha = 0.1, fill = "blue")+
  geom_line(data = df_t2, aes(x=springdates, y = 1000*Imedianv/N), color = "black", size = 1, linetype = "dotted")+geom_line(data = df_imputed2, aes(x=springdates, y = 1000*Imedianv/N), color = "black", size = 1)+
  geom_ribbon(data = df_t2, aes(x = springdates, ymin = 1000*Ilbv/N, ymax = 1000*Iubv/N), alpha = 0.1, fill = "black") + coord_fixed(50)+scale_y_log10()+
  facet_wrap(~scenario) + xlab ("Date") + ylab('Published') +theme(axis.text.x = element_text(angle=0))
plot_both_lit 
ggsave(filename = path, device = "png", dpi = 700)
path = "../out/B117_WT_both.png"
figure<-ggarrange(plot_both, plot_both_lit, ncol = 1, nrow = 2, common.legend = TRUE, legend = "right")
figure
ggsave(filename = path, device = "png", dpi = 700)
```

Below uses the Ct data (not publicly available)

Make figures with frequency of SGTF over time as well as cases colored by SGTF status
```{r}
# Make figures with frequency of SGTF over time as well as cases colored by SGTF status
path ="../out/dailySGTF.png"
plt_freq<-ggplot(Ct_data_daily_summary, aes(x = test_date, y = 100*(prev_SGTF)), color = "black") + geom_point() + geom_errorbar(aes(ymin = 100*(prev_SGTF - SE_prev), ymax = 100*(prev_SGTF+SE_prev)), width = 0.2, position = position_dodge(0.5))+
xlab ('Test Collection Date') + ylab ('Frequency (%) of cases that are S dropouts') + ggtitle('Daily frequency of S dropouts with low Ct values')
plt_freq
ggsave(filename = path, device = "png", dpi = 700) 

path = "../out/stackedbarSGTF.png"
stacked<-ggplot(Ct_data_daily_long, aes(x = test_date, y = n, fill = strain))  + geom_bar(position = "stack", stat = "identity") + xlab('Test Collection Date') + ylab('Number of low Ct positives') + theme(legend.position="top")+ scale_fill_discrete(name = "S gene presence", labels = c("S+", "S-"))
stacked
ggsave(filename = path, device = "png", dpi = 700)
```










Exploratory data analysis of the SGTF vs non-SGTF viral loads
```{r}
# Make a violin plot comparing the Ct values of side-by-side SGTF to non-SGTF
path="../out/Ct_distribs.png"
Ct_values<-ggplot(data=Ct_data, aes(factor(strain), mean_Ct), group = strain)+ geom_violin(position = position_dodge(width = 0.9)) + geom_boxplot(width = 0.1)+geom_jitter(shape = 16, alpha = 0.3, position = position_jitter(width=NULL, height = NULL))+ geom_hline(yintercept = QC_thres, linetype = "dashed") + xlab('Possible Strain') + ylab('Ct value') + ggtitle('Ct values by strain') 
Ct_values
ggsave(filename = path, device = "png", dpi = 700)

# Summary statistics
n_SGTF<-as.numeric(sum(Ct_data%>%filter(S_dropout == TRUE)%>%summarise(n())))
n_non_SGTF<-nrow(Ct_data)-n_SGTF
n_SGTF_low_Ct<-as.numeric(sum(Ct_data%>%filter(S_dropout==TRUE & mean_Ct<QC_thres)%>%summarise(n())))
n_non_SGTF_low_Ct<-as.numeric(sum(Ct_data%>%filter(S_dropout==FALSE & mean_Ct<QC_thres)%>%summarise(n())))
Ct_all_combined<-Ct_data%>%group_by(strain)%>%
  summarise(median_Ct = quantile(mean_Ct, 0.5),
    Ct_lb = quantile(mean_Ct, 0.025),
    Ct_ub = quantile(mean_Ct, 0.975),
    n_samps = n())

# Run a Kolmogorov-Smirnov test to see if there is a statistically significant difference in the Ct value distributions
ks.test(Ct_data$mean_Ct[Ct_data$strain == "SGTF"], Ct_data$mean_Ct[Ct_data$strain == "non-SGTF"])
```